src/Equations/acoustic/Model/AcousticSetup.h

```diff
@@ -0,0 +1,134 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @section DESCRIPTION
+ **/
+#ifndef ACOUSTIC_SETUP_H_
+#define ACOUSTIC_SETUP_H_
+
+#include "Model/Common.h"
+#include "Kernels/Common.h"
+#include "Numerical/Transformation.h"
+#include "generated_code/init.h"
+#include "Numerical/Eigenvalues.h"
+
+namespace seissol {
+  namespace model {
+    using Matrix44 = Eigen::Matrix<double, 4, 4>;
+
+    template<typename T>
+    inline void getTransposedCoefficientMatrix( AcousticMaterial const&  i_material,
+                                                unsigned                i_dim,
+                                                T&                      o_M )
+      {
+        o_M.setZero();
+
+        real rhoInv = 1.0 / i_material.rho;
+
+        switch (i_dim)
+          {
+            case 0:
+              o_M(1,0) = i_material.lambda;
+              o_M(0,1) = rhoInv;
+              break;
+        
+            case 1:
+              o_M(2,0) = i_material.lambda;
+              o_M(0,2) = rhoInv;
+              break;
+        
+            case 2:
+              o_M(3,0) = i_material.lambda;
+              o_M(0,3) = rhoInv;
+              break;
+              
+            default:
+              break;
+          }
+      }
+
+    template<typename Tloc, typename Tneigh>
+    inline void getTransposedGodunovState( AcousticMaterial const& local,
+                                           AcousticMaterial const& neighbor,
+                                           FaceType               faceType,
+                                           Tloc&                  QgodLocal,
+                                           Tneigh&                QgodNeighbor )
+      {
+         QgodNeighbor.setZero();
+
+         // Eigenvectors are precomputed
+         Matrix44 R = Matrix44::Zero();
+         // scale for better condition number of R
+         R(0,0) = std::sqrt(local.lambda * local.rho);
+         R(1,0) = -local.lambda;
+         R(0,1) = std::sqrt(neighbor.lambda * neighbor.rho);
+         R(1,1) = neighbor.lambda;
+         R(2,2) = local.lambda;
+         R(3,3) = local.lambda;
+
+         if (faceType == FaceType::FreeSurface) {
+          //  MaterialType materialtype = testIfAcoustic(local.mu) ? MaterialType::Acoustic : MaterialType::Elastic;
+          //  getTransposedFreeSurfaceGodunovState(materialtype, QgodLocal, QgodNeighbor, R);
```

Comment:
Does that method still need to be called here?

Feel free to extend the `getTransposedFreeSurfaceGodunovState` then (should be in the `Model` folder)—iirc there was already some code path for the pseudo-acoustic material there.